R/sensitivityCalculation.R

```diff
@@ -0,0 +1,131 @@
+#' @name sensitivityCalculation
+#' @title Carry out and visualize sensitivity analysis (with OSPSuite)
+#'
+#' @param simulation An object of type `Simulation`.
+#' @param outputPaths Path (or a vector of paths) to the output(s) for which the
+#'   sensitivity will be analyzed.
+#' @param parameterPaths A single or a vector of the parameter path(s) to be
+#'   varied.
+#' @param variationRange Optional numeric vector defining the scaling of the
+#'   parameters. The same variation range is applied to all specified
+#'   parameters. If not specified, the following vector will be used: c(0.1,
+#'   0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1, 2, 3, 4, 5,  6, 7, 8, 9, 10).
+#' @param pkParameters A vector of names of PK parameters for which the
+#'   sensitivities will be calculated. For a full set of available PK
+#'   parameters, run `names(ospsuite::StandardPKParameter)`. By default, only
+#'   the following parameters will be considered: `"C_max"`, `"t_max"`,
+#'   `"AUC_inf"`. If `NULL`, all available PK-parameters will be calculated.
+#' @param pkDataFilePath Path to excel file in which
+#'   PK-parameter data should be saved. If a file already exists, it will be
+#'   overwritten. Default is `NULL`, meaning the data will not be saved to a
+#'   spreadsheet.
+#'
+#' @examples
+#'
+#' simPath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+#' simulation <- loadSimulation(simPath)
+#' outputPaths <- "Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)"
+#' parameterPaths <- c(
+#'   "Aciclovir|Lipophilicity",
+#'   "Applications|IV 250mg 10min|Application_1|ProtocolSchemaItem|Dose",
+#'   "Neighborhoods|Kidney_pls_Kidney_ur|Aciclovir|Glomerular Filtration-GFR|GFR fraction"
+#' )
+#'
+#' # extract the results into a list of dataframes
+#' sensitivityCalculation(
+#'   simulation = simulation,
+#'   outputPaths = outputPaths,
+#'   parameterPaths = parameterPaths
+#' )
+#' @export
+sensitivityCalculation <- function(simulation,
+                                   outputPaths,
+                                   parameterPaths,
+                                   variationRange = c(seq(0.1, 1, by = 0.1), seq(2, 10, by = 1)),
+                                   pkParameters = c("C_max", "t_max", "AUC_inf"),
+                                   pkDataFilePath = NULL) {
+  # validate output paths
+  validateIsCharacter(outputPaths)
+  if (any(nchar(outputPaths) == 0L)) {
+    stop("Path name in `outputPaths` can't be an empty string.")
+  }
+
+  # validate parameter paths
+  validateIsCharacter(parameterPaths)
+  if (any(nchar(parameterPaths) == 0L)) {
+    stop("Path name in `parameterPaths` can't be an empty string.")
+  }
+
+  # validate PK parameters
+  validateIsCharacter(pkParameters)
+  if (any(nchar(pkParameters) == 0L)) {
+    stop("PK parameter name in `pkParameters` can't be an empty string.")
+  }
+  validateIsIncluded(pkParameters, names(ospsuite::StandardPKParameter))
```

Comment:
Actually just realized - this should not be checked for. There are special use cases where the user can define a "user defined PK parameter" that would have a name that is not included in `ospsuite::StandardPKParameter`

Reply:
We don't check for this anymore.